### PR TITLE
Performance & code quality optimisations

### PR DESCRIPTION
**Pull request by pascalheidmann**
Lots of smaller changes that should improve code quality as well as sometimes performance directly.
Each kind of optimisation is in its own commit to make cherrypicking & reviewing easier.

### CODE REVIEW

File: Components/Converter/XmlConverter.php [hunk 1]
```diff
@@ -113,7 +113,10 @@ public function decode($contents)
 
         foreach ($xml_values as $data) {
             unset($attributes, $value); //Remove existing values, or there will be trouble
-            extract($data); //We could use the array by itself, but this cooler.
+
+            ['type' => $type, 'value' => $value, 'level' => $level, 'tag' => $tag, 'attributes' => $attributes] = $data;
+            //extract(data); //We could use the array by itself, but this cooler.
```
Comment (mitelg) [inline Components/Converter/XmlConverter.php]: remove commented code

File: composer.json [hunk 2]
```diff
@@ -14,5 +14,9 @@
             "type": "vcs",
             "url": "https://github.com/shopwareLabs/plugin-dev-tools"
         }
-    ]
+    ],
+  "require": {
```
Comment (mitelg) [inline composer.json]: why is this necessary?

Comment (pascalheidmann) [reply composer.json]: What exactly? Adding those php extensions isn't strictly required but a recommended practice. My IDE is freaking out all the time when there are valid but not declared dependencies and composer will as well (if missing). This is therefore also already used in shopware's `composer.json`.

**Comment by mitelg** [comment]
hey @pascalheidmann 

although you closed your PR, I took some of your improvements here: b3f399cd13f579ddb70c78969949de21795f6e5c
